### PR TITLE
TP159720: Adds CI/CD Workflows

### PR DESCRIPTION
**Pull request by award28**
###
- Add CI/CD Workflows
- Implements pre-commit checks

#### Todo before Merge
- [x] #228 is merged.
- [x] Determine if the composer.json requirement of `magento/magneto-quote` is still necessary.
- [x] Perform the validation check 

## Manual Testing Steps

1.

## Pre-Submission Checklist:

- [x] You've updated the CHANGELOG following the steps [here](https://github.com/klaviyo/magento2-klaviyo#making-updates)
- [ ] **Internal Only** - If this is a release, have you:
  - [ ] Updated the links in the CHANGELOG to point towards the new versions
  - [ ] Incremented the version in the following places: module.xml and composer.json

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -14,8 +14,8 @@ Describe how you tested your change. If you are fixing a bug, please provide the
 
 - [ ] You've updated the CHANGELOG following the steps [here](https://github.com/klaviyo/magento2-klaviyo#making-updates)
 - [ ] **Internal Only** - If this is a release, have you:
-  - [ ] Updated the links in the CHANGELOG to point towards the new versions
-  - [ ] Incremented the version in the following places: module.xml and composer.json
+  - [ ] The links in the changelog have been updated to point towards the new versions
+  - [ ] The version has been incremented in the following places: module.xml and composer.json
```
Comment (chavesj) [inline .github/pull_request_template.md:18]: Line 16 wording needs to be updated to adjust for new wording of lines 17 and 18

File: .github/workflows/continuous_deployment.yml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    outputs:
+      version: ${{ steps.bump_version.outputs.version || steps.init_version.outputs.version }}
+    steps:
+    - name: Extract Branch Version
+      shell: bash
+      run: |
+        # Strip the `stable/` prefix and the `x` suffix from the current
+        # release branch
+        VERSION_PREFIX=$(echo "${GITHUB_REF#refs/heads/}" | sed -e "s/^stable\///" -e "s/x//")
+        echo "version_prefix=${VERSION_PREFIX}" >> $GITHUB_OUTPUT
```
Comment (chavesj) [inline .github/workflows/continuous_deployment.yml:21]: defining bash in a yaml file feels weird to me, should we create scripts for the bash in this file?

File: .github/pull_request_template.md (hunk 1 above)
Comment (cykolln) [reply .github/pull_request_template.md:18]: May be outside the scope of this PR, but w/ the use of changelogger should all the wording here be adjusted in general as we won't be manually updating these files anymore?

Comment (award28) [reply .github/pull_request_template.md:18]: Definitely agree with that -this should be updated. I'll do that as a part of this PR

File: .github/workflows/continuous_deployment.yml (hunk 2 above)
Comment (award28) [reply .github/workflows/continuous_deployment.yml:21]: @chavesj I think your proposal makes sense, that these should not like in this file. I'm going to create a follow up ticket in the CI/CD work to look into turning these bash scripts into reusable github actions.

**Review by chavesj — APPROVED** [review]

File: .github/workflows/continuous_deployment.yml (hunk 2 above)
Comment (award28) [reply .github/workflows/continuous_deployment.yml:21]: Ticket https://klaviyo.tpondemand.com/entity/165151-create-a-github-action-for-tag

**Pull request merged**